app/workers/emotion_selection_notification_worker.rb

```diff
@@ -6,16 +6,66 @@ def initialize
     @time_period = TimePeriod.find_or_create_time_period
   end
 
-  def run_notification
-    return unless Date.current.strftime('%A').casecmp?(ENV.fetch('DAY_TO_SEND_INVITES'))
+  def run_notification(operation = nil)
+    case operation
+    when :send_results_email
+      return unless Date.current.strftime('%A').casecmp?(ENV.fetch('DAY_TO_SEND_RESULTS_EMAIL'))
 
-    time_period.update(due_date: Date.current)
-    run_notification!
+      run_results_email!
+    else
+      return unless Date.current.strftime('%A').casecmp?(ENV.fetch('DAY_TO_SEND_INVITES'))
+
+      time_period.update(due_date: Date.current)
+      run_notification!
+    end
   end
```

Comment:
@LyubovP even if you want to use the same worker it's better to write separate method for each notification and please add test coverage for you code